### PR TITLE
WaveformExtractor.select_units also functional if  we.has_recording()=False

### PR DESCRIPTION
**Pull request by LiesDec**
This is a pull request following up on issue #2281.
The proposed changes enable the method WaveformExtractor.select_units to also work if the WaveformExtractor had not recording.

### CODE REVIEW

**Comment by LiesDec** [comment]
The fix in the second commit was necessary for me (and potentially other users) because in the lab we store our data on a NAS sinology data server. These create @eaDir files to handle user specific views. These files need to be skipped when walking through a directory like what happens in [ _load_extension_data](https://github.com/SpikeInterface/spikeinterface/blob/main/src/spikeinterface/core/waveform_extractor.py#L1973).

File: src/spikeinterface/core/waveform_extractor.py [hunk 1]
```diff
@@ -1990,6 +1994,9 @@ def _load_extension_data(self):
                     ext_data = pd.read_csv(ext_data_file, index_col=0)
                 elif ext_data_file.suffix == ".pkl":
                     ext_data = pickle.load(ext_data_file.open("rb"))
+                else:
+                    print("{} not created by spikeinterface, skipping this file".format(ext_data_file.name))
```
Comment (alejoe91) [inline src/spikeinterface/core/waveform_extractor.py:1998]:
```suggestion
                    print(f"{ext_data_file.name} not created by spikeinterface, skipping this file")
```
We prefer to use f strings throughout the code!

File: src/spikeinterface/core/waveform_extractor.py [hunk 2]
```diff
@@ -800,6 +800,9 @@ def select_units(self, unit_ids, new_folder=None, use_relative_path: bool = Fals
 
                 if self.has_recording():
                     self.recording.dump(new_folder / "recording.json", relative_to=relative_to)
+                else:
+                    shutil.copytree(self.folder / "recording_info", new_folder / "recording_info")
```
Comment (samuelgarcia) [inline src/spikeinterface/core/waveform_extractor.py:804]: I would remove the else. We should copy this file for all situation no ?

File: src/spikeinterface/core/waveform_extractor.py [hunk 3]
```diff
@@ -1990,6 +1994,9 @@ def _load_extension_data(self):
                     ext_data = pd.read_csv(ext_data_file, index_col=0)
                 elif ext_data_file.suffix == ".pkl":
                     ext_data = pickle.load(ext_data_file.open("rb"))
+                else:
+                    print(f"{ext_data_file.name} not created by spikeinterface, skipping this file")
+                    continue
```
Comment (samuelgarcia) [inline src/spikeinterface/core/waveform_extractor.py:1999]: Not sure to see the use case.
Why would there be a file not created by spikeinterface is this subfolder ?

**Comment by samuelgarcia** [comment]
Hi @LiesDec.
Merci beaucoup.
I made some coments.

**Comment by alejoe91** [comment]
> Hi @LiesDec. Merci beaucoup. I made some coments.

BEcause of this: https://github.com/SpikeInterface/spikeinterface/pull/2368#issuecomment-1866194157

File: src/spikeinterface/core/waveform_extractor.py (hunk 3 above)
Comment (samuelgarcia) [reply src/spikeinterface/core/waveform_extractor.py:1999]: oups sorry. I haven't read in detail the comments.

File: src/spikeinterface/core/waveform_extractor.py (hunk 2 above)
Comment (LiesDec) [reply src/spikeinterface/core/waveform_extractor.py:804]: @samuelgarcia I thought the file is copied when you dump the recording?

Comment (alejoe91) [reply src/spikeinterface/core/waveform_extractor.py:804]: I agree with @samuelgarcia 

Please remove the else and copy recording info in both cases (with and without recording)

Comment (LiesDec) [reply src/spikeinterface/core/waveform_extractor.py]: @samuelgarcia @alejoe91  Allright, done :)

File: src/spikeinterface/core/waveform_extractor.py [hunk 4]
```diff
@@ -1990,6 +1994,9 @@ def _load_extension_data(self):
                     ext_data = pd.read_csv(ext_data_file, index_col=0)
                 elif ext_data_file.suffix == ".pkl":
                     ext_data = pickle.load(ext_data_file.open("rb"))
+                else:
+                    print(f"{ext_data_file.name} not created by spikeinterface, skipping this file")
```
Comment (alejoe91) [inline src/spikeinterface/core/waveform_extractor.py:1998]:
```suggestion
```
Removing the print but keeping the logic!

**Review by alejoe91 — APPROVED** [review]

**Review by samuelgarcia — APPROVED** [review]

**Pull request merged**